File: medplum-java/medplum-model/src/main/java/com/fhir/CapabilityStatement_Resource.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCapabilityStatement_Resource.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface CapabilityStatement_Resource {
  Optional<List<Extension>> extension();

  Optional<Capabilitystatement_resourceConditionalread> conditionalRead();

  Optional<List<CapabilityStatement_SearchParam>> searchParam();

  Optional<List<Extension>> modifierExtension();

  Optional<Boolean> conditionalCreate();

  Optional<Capabilitystatement_resourceConditionaldelete> conditionalDelete();

  Optional<List<String>> searchInclude();

  Optional<String> id();

  Optional<canonical> profile();

  Optional<Capabilitystatement_resourceVersioning> versioning();

  Optional<markdown> documentation();

  Optional<List<CapabilityStatement_Interaction>> interaction();

  Optional<List<canonical>> supportedProfile();

  Optional<List<Capabilitystatement_resourceReferencepolicyItem>> referencePolicy();

  Optional<List<String>> searchRevInclude();

  Optional<List<CapabilityStatement_Operation>> operation();

  Optional<Boolean> conditionalUpdate();

  Optional<code> type();

  Optional<Boolean> readHistory();

  Optional<Boolean> updateCreate();

  static ImmutableCapabilityStatement_Resource.Builder builder() {
    return ImmutableCapabilityStatement_Resource.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/CapabilityStatement_Resource.ts
import { canonical } from "./canonical";
import { CapabilityStatement_Interaction } from "./CapabilityStatement_Interaction";
import { CapabilityStatement_Operation } from "./CapabilityStatement_Operation";
import { Capabilitystatement_resourceConditionaldelete } from "./Capabilitystatement_resourceConditionaldelete";
import { Capabilitystatement_resourceConditionalread } from "./Capabilitystatement_resourceConditionalread";
import { Capabilitystatement_resourceReferencepolicyItem } from "./Capabilitystatement_resourceReferencepolicyItem";
import { Capabilitystatement_resourceVersioning } from "./Capabilitystatement_resourceVersioning";
import { CapabilityStatement_SearchParam } from "./CapabilityStatement_SearchParam";
import { code } from "./code";
import { Extension } from "./Extension";
import { markdown } from "./markdown";

export interface CapabilityStatement_Resource {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: code | null | undefined;
    profile: canonical | null | undefined;
    supportedProfile: canonical[] | null | undefined;
    documentation: markdown | null | undefined;
    interaction: CapabilityStatement_Interaction[] | null | undefined;
    versioning: Capabilitystatement_resourceVersioning | null | undefined;
    readHistory: boolean | null | undefined;
    updateCreate: boolean | null | undefined;
    conditionalCreate: boolean | null | undefined;
    conditionalRead:
        | Capabilitystatement_resourceConditionalread
        | null
        | undefined;
    conditionalUpdate: boolean | null | undefined;
    conditionalDelete:
        | Capabilitystatement_resourceConditionaldelete
        | null
        | undefined;
    referencePolicy:
        | Capabilitystatement_resourceReferencepolicyItem[]
        | null
        | undefined;
    searchInclude: string[] | null | undefined;
    searchRevInclude: string[] | null | undefined;
    searchParam: CapabilityStatement_SearchParam[] | null | undefined;
    operation: CapabilityStatement_Operation[] | null | undefined;
}
